tests/e2e/prevent_csi_session_leaks.go

```diff
@@ -0,0 +1,174 @@
+		framework.ExpectNoError(err, "Unable to find ready and schedulable Node")
+		if !(len(nodeList.Items) > 0) {
+			framework.Failf("Unable to find ready and schedulable Node")
+		}
+		// Get CSI Controller's replica count from the setup
+		controllerClusterConfig := os.Getenv(contollerClusterKubeConfig)
+		c = client
+		if controllerClusterConfig != "" {
+			framework.Logf("Creating client for remote kubeconfig")
+			remoteC, err := createKubernetesClientFromConfig(controllerClusterConfig)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			c = remoteC
+		}
+		deployment, err = c.AppsV1().Deployments(csiSystemNamespace).Get(ctx,
+			vSphereCSIControllerPodNamePrefix, metav1.GetOptions{})
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		csiReplicaCount = *deployment.Spec.Replicas
+	})
+
+	ginkgo.It("Verify session creted with unique user agent", func() {
+		var cancel context.CancelFunc
+		ctx, cancel := context.WithCancel(context.Background())
+		defer cancel()
+		// check sessions are uniquely identifiable
+		tasksListViewEnabled := isCsiFssEnabled(ctx, client, GetAndExpectStringEnvVar(envCSINamespace),
+			"listview-tasks")
+		gomega.Expect(tasksListViewEnabled).To(gomega.BeTrue())
+		sessionList := getSessionList(ctx)
+		count := 0
+		matchRgx := "k8s-csi-useragent-[^.]+-[^.]+"
+		rgx, err := regexp.Compile(matchRgx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, session := range sessionList {
+			matched := rgx.MatchString(session.UserAgent)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			if matched {
+				count++
+			}
+		}
+		gomega.Expect(count).NotTo(gomega.Equal(0))
+	})
+
+	ginkgo.It("Verify session cleaned up on termination", func() {
+		var cancel context.CancelFunc
+		ctx, cancel := context.WithCancel(context.Background())
+		ignoreLabels := make(map[string]string)
+		defer cancel()
+
+		defer func() {
+			// start CSI driver
+			framework.Logf("Starting CSI driver")
+			err := updateDeploymentReplicawithWait(c, csiReplicaCount, vSphereCSIControllerPodNamePrefix,
+				csiSystemNamespace)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+
+			// Wait for the CSI Pods to be up and Running
+			list_of_pods, err := fpod.GetPodsInNamespace(client, csiSystemNamespace, ignoreLabels)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+			num_csi_pods := len(list_of_pods)
+			err = fpod.WaitForPodsRunningReady(client, csiSystemNamespace, int32(num_csi_pods), 0,
+				pollTimeout, ignoreLabels)
+			gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		}()
+
+		sessionList := getSessionList(ctx)
+		count := 0
+		matchRgx := "k8s-csi-useragent-[^.]+"
+		rgx, err := regexp.Compile(matchRgx)
+		gomega.Expect(err).NotTo(gomega.HaveOccurred())
+		for _, session := range sessionList {
+			matched := rgx.MatchString(session.UserAgent)
+			if matched {
+				count++
+			}
+		}
+		framework.Logf("Stoping CSI driver")
+		err = updateDeploymentReplicawithWait(c, 0, vSphereCSIControllerPodNamePrefix,
```

Comment:
we have `stopCSIPods()` and `startCSIPods()` in utils which already does the task and before we stop csi pods we need to collect logs from them (which is already part of `stopCSIPods()`)

Reply:
replacing code with these available functions